libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/MemoryMessageBus.scala

```diff
@@ -0,0 +1,39 @@
+package com.advancedtelematic.libats.messaging
+
+import akka.NotUsed
+import akka.actor.ActorRef
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.Source
+import com.advancedtelematic.libats.messaging.Messages.MessageLike
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class MemoryMessageBus {
+  private val _buffer = mutable.Queue.empty[Any]
+  private var _receiver = Option.empty[ActorRef]
+
+  def subscribe[T]()(implicit m: MessageLike[T]): Source[T, NotUsed] = {
+    Source.actorRef(MessageBus.DEFAULT_CLIENT_BUFFER_SIZE, OverflowStrategy.dropTail).mapMaterializedValue { ref =>
+      _receiver = Some(ref)
```

Comment:
So `subscribe`-ing twice will make it so that the last one gets all the messages? I suppose that this class is only supposed to be used with a single subscribe (testing purposes?) so it is probably no big deal.